src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -319,6 +321,14 @@ public void insertRecordToBuffer(SinkRecord kafkaSinkRecord) {
    */
   private boolean shouldIgnoreAddingRecordToBuffer(SinkRecord kafkaSinkRecord) {
     if (this.isOffsetResetInKafka.get()) {
+      if (offsetPersistedInSnowflake.get() == -1L && kafkaSinkRecord.kafkaOffset() == initKafkaOffset) {
+        LOGGER.info(
+                "First insert to table, get the desired init offset:{} from Kafka, we can add this offset to buffer for channel:{}",
```

Comment:
`First insert to table, first seen offset from Kafka:{}, we can add this offset to buffer for channel:{}",`